src/components/LocaleContextProvider.tsx

```diff
@@ -0,0 +1,149 @@
+import React, {createContext, useMemo} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import {ValueOf} from 'type-fest';
+import ONYXKEYS from '../ONYXKEYS';
+import * as Localize from '../libs/Localize';
+import DateUtils from '../libs/DateUtils';
+import * as NumberFormatUtils from '../libs/NumberFormatUtils';
+import * as LocaleDigitUtils from '../libs/LocaleDigitUtils';
+import CONST from '../CONST';
+import TIMEZONES from '../TIMEZONES';
+import compose from '../libs/compose';
+import withCurrentUserPersonalDetails from './withCurrentUserPersonalDetails';
+import * as LocalePhoneNumber from '../libs/LocalePhoneNumber';
+
+type SelectedTimezone = {
+    /** Value of the selected timezone */
+    selected: (typeof TIMEZONES)[number];
+};
+
+type CurrentUserPersonalDetails = {
+    /** Timezone of the current user */
+    timezone?: SelectedTimezone;
+};
```

Comment:
```suggestion
type CurrentUserPersonalDetails = Pick<PersonalDetails, 'timezone'>;
```

Shouldn't this be easier?